src/CommunityCommons/javasource/communitycommons/StringUtils.java

```diff
@@ -371,22 +371,47 @@ public static String decryptString(String key, String valueToDecrypt) throws Exc
 		return new String(c.doFinal(encryptedData));
 	}
 
-	public static String generateHmacSha256Hash(String key, String valueToEncrypt)
+	private static byte[] generateHmacSha256Bytes(String key, String valueToEncrypt)
 	{
 		try {
 			SecretKeySpec secretKey = new SecretKeySpec(key.getBytes("UTF-8"), "HmacSHA256");
 			Mac mac = Mac.getInstance("HmacSHA256");
 			mac.init(secretKey);
 			mac.update(valueToEncrypt.getBytes("UTF-8"));
 			byte[] hmacData = mac.doFinal();
-
-            return Base64.getEncoder().encodeToString(hmacData);
+			
+			return hmacData;
 		}
 		catch (UnsupportedEncodingException | IllegalStateException | InvalidKeyException | NoSuchAlgorithmException e) {
+			throw new RuntimeException(e.getMessage(), e);
+		}
+	}
+	
+	public static String generateHmacSha256(String key, String valueToEncrypt)
+	{
+		try {
+            byte[] hash = generateHmacSha256Bytes(key, valueToEncrypt);
+    		StringBuffer result = new StringBuffer();
+    		for (byte b : hash) {
+    			result.append(String.format("%02x", b));
+    		}
+    		return result.toString();
+		}
+		catch (RuntimeException e) {
 			throw new RuntimeException("CommunityCommons::EncodeHmacSha256::Unable to encode: " + e.getMessage(), e);
 		}
 	}
 
+	public static String generateHmacSha256Hash(String key, String valueToEncrypt)
+	{
```

Comment:
Why is this method needed, when only `generateHmacSha256()` is exposed as a Java action? Or am I missing something?

Reply:
This is the existing functionality which exposed the Base64 encoding. This needs to be retained, and the new functionality will expose the hex encoded version.